mongo/bulk_write.go

```diff
@@ -430,6 +432,9 @@ type updateDoc struct {
 func (doc updateDoc) marshal(bsonOpts *options.BSONOptions, registry *bson.Registry) (bsoncore.Document, error) {
 	f, err := marshal(doc.filter, bsonOpts, registry)
 	if err != nil {
+		if doc.filter == nil {
+			return nil, fmt.Errorf("%w: filter is required", err)
+		}
```

Comment:
update the error message when the filter is not given.

Reply:
Is there a reason to nest this nil filter check in this error handling block? Can we move this check to the beginning of `marshal`?

Optional: Consider a clearer error message, like
> update filter cannot be nil